server/src/main/java/org/elasticsearch/index/mapper/DocumentParserContext.java

```diff
@@ -260,7 +265,10 @@ public final Collection<String> getIgnoredFields() {
      * Add the given ignored values to the corresponding list.
      */
     public final void addIgnoredField(IgnoredSourceFieldMapper.NameValue values) {
-        ignoredFieldValues.add(values);
+        if (sourceStored == false) {
```

Comment:
I wonder whether this if check needed here? This is also checked in `parseNonDynamicArray(...)` and so this should always be false?

Reply:
There may be other reasons where we try to store a subfield, e.g. a disabled object. Double-storing is redundant, though it has no side effects.

Reply:
Gotcha, this makes sense to me now.